// @draftbit/maps/lib/module/index.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import MapView from 'react-native-maps';
export class MapSimple extends React.PureComponent {
  render() {
    const {
      aspectRatio,
      initialLatitude,
      initialLongitude,
      initialLatitudeDelta,
      initialLongitudeDelta,
      markers,
      style,
      pinColor,
      interactionDisabled
    } = this.props;
    return /*#__PURE__*/React.createElement(MapView, {
      initialRegion: {
        latitude: initialLatitude,
        longitude: initialLongitude,
        latitudeDelta: initialLatitudeDelta,
        longitudeDelta: initialLongitudeDelta
      },
      pitchEnabled: !interactionDisabled,
      rotateEnabled: !interactionDisabled,
      scrollEnabled: !interactionDisabled,
      zoomEnabled: !interactionDisabled,
      style: [{
        width: '100%',
        aspectRatio
      }, style]
    }, markers.map((m, i) => /*#__PURE__*/React.createElement(MapView.Marker, {
      coordinate: {
        latitude: m.latitude,
        longitude: m.longitude
      },
      title: m.title,
      description: m.description,
      pinColor: pinColor,
      key: i
    })));
  }

}

_defineProperty(MapSimple, "defaultProps", {
  aspectRatio: 1,
  markers: []
});
