// @draftbit/maps/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MapSimple = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNativeMaps = _interopRequireDefault(require("react-native-maps"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class MapSimple extends _react.default.PureComponent {
  render() {
    const {
      aspectRatio,
      initialLatitude,
      initialLongitude,
      initialLatitudeDelta,
      initialLongitudeDelta,
      markers,
      style,
      pinColor,
      interactionDisabled
    } = this.props;
    return /*#__PURE__*/_react.default.createElement(_reactNativeMaps.default, {
      initialRegion: {
        latitude: initialLatitude,
        longitude: initialLongitude,
        latitudeDelta: initialLatitudeDelta,
        longitudeDelta: initialLongitudeDelta
      },
      pitchEnabled: !interactionDisabled,
      rotateEnabled: !interactionDisabled,
      scrollEnabled: !interactionDisabled,
      zoomEnabled: !interactionDisabled,
      style: [{
        width: '100%',
        aspectRatio
      }, style]
    }, markers.map((m, i) => /*#__PURE__*/_react.default.createElement(_reactNativeMaps.default.Marker, {
      coordinate: {
        latitude: m.latitude,
        longitude: m.longitude
      },
      title: m.title,
      description: m.description,
      pinColor: pinColor,
      key: i
    })));
  }

}

exports.MapSimple = MapSimple;

_defineProperty(MapSimple, "defaultProps", {
  aspectRatio: 1,
  markers: []
});
