// @draftbit/core/src/components/Surface.js
/* Copied from https://github.com/callstack/react-native-paper/blob/main/src/components/Surface.tsx */
import * as React from "react";
import { Animated, StyleSheet, View, } from "react-native";
import shadow from "../styles/shadow";
import overlay from "../styles/overlay";
import { withTheme } from "../theming";
import { COMPONENT_TYPES, createElevationType } from "@draftbit/types";
const Surface = ({ elevation, style, theme, children, ...rest }) => {
    const { elevation: styleElevation = 3, borderRadius: radius } = (StyleSheet.flatten(style) || {});
    const { dark: isDarkTheme, mode, colors } = theme;
    const borderRadius = radius || theme.roundness;
    const ele = elevation || styleElevation;
    return (React.createElement(Animated.View, { ...rest, style: [
            {
                minHeight: 40,
                borderRadius,
                backgroundColor: isDarkTheme && mode === "adaptive"
                    ? overlay(ele, colors.surface)
                    : colors.surface,
            },
            elevation ? shadow(elevation) : null,
            style,
        ] },
        React.createElement(View, { style: { overflow: "hidden", borderRadius } }, children)));
};
export default withTheme(Surface);
export const SEED_DATA = {
    name: "Surface",
    tag: "Surface",
    description: "An elevated container",
    category: COMPONENT_TYPES.layout,
    layout: {},
    props: {
        elevation: createElevationType(0),
    },
};
