// @draftbit/core/lib/module/components/CircleImage.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from "react";
import { Image } from "react-native";
import Config from "./Config";
import { GROUPS, COMPONENT_TYPES, FORM_TYPES, PROP_TYPES, createImageProp } from "@draftbit/types";

const CircleImage = ({
  source = Config.placeholderImageURL,
  size = 60,
  style,
  ...props
}) => {
  const borderRadius = size / 2;
  return /*#__PURE__*/React.createElement(Image, _extends({
    style: [{
      width: size,
      height: size,
      borderRadius
    }, style],
    source: typeof source === "string" ? {
      uri: source
    } : source,
    resizeMode: "cover"
  }, props));
};

export default CircleImage;
export const SEED_DATA = {
  name: "Circle Image",
  tag: "CircleImage",
  description: "A circle image",
  category: COMPONENT_TYPES.media,
  props: {
    source: createImageProp(),
    size: {
      group: GROUPS.basic,
      label: "Size",
      description: "Size of your circle image",
      editable: true,
      required: false,
      formType: FORM_TYPES.number,
      propType: PROP_TYPES.NUMBER,
      min: 0,
      max: 300,
      precision: 0,
      step: 1,
      defaultValue: 60
    }
  }
};
