// @draftbit/maps/src/index.tsx
import React from 'react';
import MapView from 'react-native-maps';

export class MapSimple extends React.PureComponent {
  static defaultProps = {
    aspectRatio: 1,
    markers: [],
  };

  render() {
    const {
      aspectRatio,
      initialLatitude,
      initialLongitude,
      initialLatitudeDelta,
      initialLongitudeDelta,
      markers,
      style,
      pinColor,
      interactionDisabled,
    } = this.props;

    return (
      <MapView
        initialRegion={{
          latitude: initialLatitude,
          longitude: initialLongitude,
          latitudeDelta: initialLatitudeDelta,
          longitudeDelta: initialLongitudeDelta,
        }}
        pitchEnabled={!interactionDisabled}
        rotateEnabled={!interactionDisabled}
        scrollEnabled={!interactionDisabled}
        zoomEnabled={!interactionDisabled}
        style={[
          {
            width: '100%',
            aspectRatio,
          },
          style,
        ]}
      >
        {markers.map((m, i) => (
          <MapView.Marker
            coordinate={{
              latitude: m.latitude,
              longitude: m.longitude,
            }}
            title={m.title}
            description={m.description}
            pinColor={pinColor}
            key={i}
          />
        ))}
      </MapView>
    );
  }
}
