// @draftbit/core/src/components/Checkbox/CheckboxGroup.js
import * as React from "react";
import { View } from "react-native";
import { GROUPS, COMPONENT_TYPES, FORM_TYPES, createTextProp, PROP_TYPES, createFieldNameProp, createDirectionProp, } from "@draftbit/types";
import { checkboxGroupContext, Direction } from "./context";
const { Provider } = checkboxGroupContext;
const CheckboxGroup = ({ direction = Direction.Vertical, values, onValueChange = () => { }, style, children, ...rest }) => {
    const _containerStyle = [
        {
            flexDirection: direction === Direction.Horizontal ? "row" : "column",
            overflow: "hidden",
        },
    ];
    if (direction !== Direction.Vertical) {
        _containerStyle.push({
            alignItems: "center",
        });
    }
    return (React.createElement(View, { style: [{ minHeight: 40 }, style], ...rest },
        React.createElement(Provider, { value: { values, onValueChange, direction } },
            React.createElement(View, { style: _containerStyle }, children))));
};
export default CheckboxGroup;
export const SEED_DATA = {
    name: "Checkbox Group",
    tag: "CheckboxGroup",
    category: COMPONENT_TYPES.deprecated,
    layout: {},
    props: {
        direction: createDirectionProp(),
        values: createTextProp({
            group: GROUPS.data,
            formType: FORM_TYPES.flatArray,
            propType: PROP_TYPES.ARRAY,
            label: "Values",
            description: "Currently selected values of the checkbox group",
            required: true,
            defaultValue: null,
        }),
        fieldName: createFieldNameProp({
            defaultValue: "checkboxGroupValue",
            handlerPropName: "onValueChange",
            valuePropName: "checkboxGroupValue",
        }),
    },
};
