// @draftbit/core/lib/module/components/Checkbox/CheckboxGroup.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from "react";
import { View } from "react-native";
import { GROUPS, COMPONENT_TYPES, FORM_TYPES, createTextProp, PROP_TYPES, createFieldNameProp, createDirectionProp } from "@draftbit/types";
import { checkboxGroupContext, Direction } from "./context";
const {
  Provider
} = checkboxGroupContext;

const CheckboxGroup = ({
  direction = Direction.Vertical,
  values,
  onValueChange = () => {},
  style,
  children,
  ...rest
}) => {
  const _containerStyle = [{
    flexDirection: direction === Direction.Horizontal ? "row" : "column",
    overflow: "hidden"
  }];

  if (direction !== Direction.Vertical) {
    _containerStyle.push({
      alignItems: "center"
    });
  }

  return /*#__PURE__*/React.createElement(View, _extends({
    style: [{
      minHeight: 40
    }, style]
  }, rest), /*#__PURE__*/React.createElement(Provider, {
    value: {
      values,
      onValueChange,
      direction
    }
  }, /*#__PURE__*/React.createElement(View, {
    style: _containerStyle
  }, children)));
};

export default CheckboxGroup;
export const SEED_DATA = {
  name: "Checkbox Group",
  tag: "CheckboxGroup",
  category: COMPONENT_TYPES.deprecated,
  layout: {},
  props: {
    direction: createDirectionProp(),
    values: createTextProp({
      group: GROUPS.data,
      formType: FORM_TYPES.flatArray,
      propType: PROP_TYPES.ARRAY,
      label: "Values",
      description: "Currently selected values of the checkbox group",
      required: true,
      defaultValue: null
    }),
    fieldName: createFieldNameProp({
      defaultValue: "checkboxGroupValue",
      handlerPropName: "onValueChange",
      valuePropName: "checkboxGroupValue"
    })
  }
};
