// @draftbit/core/lib/commonjs/components/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = exports.Link = exports.BaseLink = exports.ButtonOutline = exports.ButtonSolid = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _theming = require("../theming");

var _types = require("@draftbit/types");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const CONSTANTS = {
  baseHeight: 42,
  borderRadius: 4,
  padding: 8,
  icon: 24
};

function Base({
  Icon,
  icon,
  title,
  onPress,
  loading,
  disabled,
  style,
  ...props
}) {
  const {
    color,
    fontFamily,
    fontWeight,
    fontSize,
    lineHeight,
    letterSpacing,
    textTransform,
    textAlign,
    textDecorationLine,
    textDecorationColor,
    textDecorationStyle,
    ...buttonStyles
  } = _reactNative.StyleSheet.flatten(style || {});

  const titleStyles = {
    color,
    fontFamily,
    fontWeight,
    fontSize,
    lineHeight,
    letterSpacing,
    textTransform,
    textAlign,
    textDecorationLine,
    textDecorationColor,
    textDecorationStyle
  };

  if (textAlign === "left") {
    buttonStyles.justifyContent = "flex-start";
  }

  if (textAlign === "right") {
    buttonStyles.justifyContent = "flex-end";
  }

  return /*#__PURE__*/React.createElement(_reactNative.Pressable, _extends({
    onPress: onPress,
    disabled: disabled || loading,
    style: ({
      pressed
    }) => {
      return [styles.base, {
        opacity: pressed || disabled ? 0.75 : 1
      }, buttonStyles];
    }
  }, props), loading ? /*#__PURE__*/React.createElement(_reactNative.ActivityIndicator, {
    size: "small",
    color: color,
    style: styles.loading
  }) : null, icon && !loading ? /*#__PURE__*/React.createElement(Icon, {
    name: icon,
    color: color,
    style: styles.icon,
    size: CONSTANTS.icon
  }) : null, /*#__PURE__*/React.createElement(_reactNative.Text, {
    style: titleStyles
  }, title));
}

const Solid = ({
  style,
  theme,
  ...props
}) => {
  return /*#__PURE__*/React.createElement(Base, _extends({
    style: [{
      color: "#FFF",
      borderRadius: theme.roundness,
      backgroundColor: theme.colors.primary
    }, style]
  }, props));
};

const ButtonSolid = (0, _theming.withTheme)(Solid);
exports.ButtonSolid = ButtonSolid;

const Outline = ({
  style,
  theme,
  ...props
}) => {
  return /*#__PURE__*/React.createElement(Base, _extends({
    style: [styles.outline, {
      borderRadius: theme.roundness,
      borderColor: theme.colors.primary,
      color: theme.colors.primary
    }, style]
  }, props));
};

const ButtonOutline = (0, _theming.withTheme)(Outline);
exports.ButtonOutline = ButtonOutline;

const BaseLink = ({
  style,
  theme,
  ...props
}) => {
  return /*#__PURE__*/React.createElement(Base, _extends({
    style: [styles.bare, {
      color: theme.colors.primary
    }, style],
    hitSlop: 8
  }, props));
};

exports.BaseLink = BaseLink;
const Link = (0, _theming.withTheme)(BaseLink);
exports.Link = Link;

const styles = _reactNative.StyleSheet.create({
  base: {
    position: "relative",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    minHeight: CONSTANTS.baseHeight,
    paddingHorizontal: 12,
    fontFamily: "System",
    fontWeight: "700",
    ..._reactNative.Platform.select({
      web: {
        cursor: "pointer",
        userSelect: "none"
      }
    })
  },
  outline: {
    backgroundColor: "transparent",
    borderWidth: 1
  },
  bare: {
    backgroundColor: "transparent",
    padding: 0,
    minHeight: undefined
  },
  loading: {
    marginRight: 6
  },
  icon: { ..._reactNative.Platform.select({
      web: {
        marginTop: 1,
        marginRight: 4,
        alignSelf: "center"
      },
      default: {
        marginBottom: 2,
        marginRight: 4,
        alignSelf: "center"
      }
    })
  }
});

const SEED_DATA_PROPS = {
  icon: (0, _types.createIconProp)({
    defaultValue: null,
    required: false
  }),
  title: (0, _types.createTextProp)({
    label: "Label",
    description: "Button Label",
    defaultValue: "Get Started"
  }),
  disabled: (0, _types.createBoolProp)({
    group: _types.GROUPS.basic,
    label: "Disabled",
    description: "Whether the button should be disabled"
  }),
  loading: (0, _types.createBoolProp)({
    group: _types.GROUPS.basic,
    label: "Loading",
    description: "Whether to show a loading indicator"
  }),
  onPress: (0, _types.createActionProp)()
};
const LAYOUT = {
  backgroundColor: "transparent",
  borderRadius: 8,
  fontFamily: "system-700"
};
const SEED_DATA = [{
  name: "Button Outline",
  tag: "ButtonOutline",
  category: _types.COMPONENT_TYPES.button,
  layout: { ...LAYOUT,
    backgroundColor: "transparent",
    borderWidth: 1,
    textAlign: "center"
  },
  props: SEED_DATA_PROPS
}, {
  name: "Button Solid",
  tag: "ButtonSolid",
  category: _types.COMPONENT_TYPES.button,
  layout: { ...LAYOUT,
    backgroundColor: "primary",
    textAlign: "center"
  },
  props: SEED_DATA_PROPS
}, {
  name: "Link",
  tag: "Link",
  category: _types.COMPONENT_TYPES.button,
  layout: { ...LAYOUT,
    backgroundColor: "transparent",
    color: "primary",
    padding: 0,
    minHeight: undefined
  },
  props: SEED_DATA_PROPS
}];
exports.SEED_DATA = SEED_DATA;
