// @draftbit/core/lib/commonjs/components/Divider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _theming = require("../theming");

var _types = require("@draftbit/types");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Divider = ({
  style,
  color,
  height,
  theme: {
    colors
  },
  ...rest
}) => {
  return /*#__PURE__*/React.createElement(_reactNative.View, _extends({
    style: [{
      backgroundColor: color || colors.divider
    }, style, {
      height: height || _reactNative.StyleSheet.hairlineWidth
    }]
  }, rest));
};

var _default = (0, _theming.withTheme)(Divider);

exports.default = _default;
const SEED_DATA = [{
  name: "Divider",
  tag: "Divider",
  category: _types.COMPONENT_TYPES.layout,
  description: "A horizontal line used to divide content",
  preview_image_url: "{CLOUDINARY_URL}/Divider.png",
  supports_list_render: false,
  props: {
    color: {
      group: _types.GROUPS.basic,
      label: "Color",
      description: "The color of the divider",
      editable: true,
      required: false,
      defaultValue: "divider",
      formType: _types.FORM_TYPES.color,
      propType: _types.PROP_TYPES.THEME
    },
    height: {
      group: _types.GROUPS.basic,
      label: "Height",
      description: "The height (thickness) of the divider. If not provided, defaults to StyleSheet.hairlineWidth.",
      editable: true,
      required: false,
      formType: _types.FORM_TYPES.number,
      propType: _types.PROP_TYPES.NUMBER,
      defaultValue: 1,
      min: 1,
      max: 50,
      step: 1,
      precision: 0
    }
  },
  layout: {
    height: 1
  }
}];
exports.SEED_DATA = SEED_DATA;
