// @draftbit/core/lib/commonjs/components/HeaderOverline.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _theming = require("../theming");

var _types = require("@draftbit/types");

var _Header = _interopRequireDefault(require("./Header"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const HeaderOverline = ({
  Icon,
  title,
  buttonText,
  icon,
  onPress = () => {},
  style,
  theme: {
    colors,
    typography
  }
}) => {
  return /*#__PURE__*/React.createElement(_Header.default, {
    Icon: Icon,
    titleTypeStyle: typography.overline,
    titleColor: colors.light,
    title: title && title.toUpperCase(),
    buttonText: buttonText,
    icon: icon,
    dividerTopMargin: 12,
    onPress: onPress,
    style: style
  });
};

var _default = (0, _theming.withTheme)(HeaderOverline);

exports.default = _default;
const SEED_DATA = [{
  name: "Header Overline",
  tag: "HeaderOverline",
  description: "A small header with an optional touchable right aligned text and icon.",
  category: _types.COMPONENT_TYPES.header,
  preview_image_url: "{CLOUDINARY_URL}/Header_OverlineIconText.png",
  supports_list_render: false,
  props: {
    title: {
      group: _types.GROUPS.data,
      label: "Title",
      description: "Text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "Title",
      editable: true,
      required: false
    },
    buttonText: {
      group: _types.GROUPS.data,
      label: "Button text",
      description: "Right aligned button text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "See All",
      editable: true,
      required: false
    },
    icon: {
      group: _types.GROUPS.basic,
      label: "Icon",
      description: "Name of icon to display",
      formType: _types.FORM_TYPES.icon,
      propType: _types.PROP_TYPES.ASSET,
      defaultValue: null,
      editable: true,
      required: false
    },
    onPress: {
      group: _types.GROUPS.basic,
      label: "Action",
      description: "Action to execute when button pressed",
      editable: true,
      required: false,
      formType: _types.FORM_TYPES.action,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: null
    }
  },
  layout: {}
}];
exports.SEED_DATA = SEED_DATA;
