// @draftbit/core/lib/commonjs/components/RowHeadlineImageCaption.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _theming = require("../theming");

var _types = require("@draftbit/types");

var _Row = _interopRequireDefault(require("./Row"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const RowHeadlineImageCaption = ({
  title,
  subtitle,
  caption,
  image,
  style,
  theme: {
    colors,
    typography
  }
}) => {
  return /*#__PURE__*/React.createElement(_Row.default, {
    titleTypeStyle: typography.headline6,
    titleColor: colors.strong,
    subtitleTypeStyle: typography.body2,
    subtitleColor: colors.medium,
    title: title,
    subtitle: subtitle,
    image: image,
    right: () => /*#__PURE__*/React.createElement(_reactNative.Text, {
      style: { ...typography.caption,
        color: colors.strong,
        marginLeft: 16
      }
    }, caption),
    style: style
  });
};

var _default = (0, _theming.withTheme)(RowHeadlineImageCaption);

exports.default = _default;
const SEED_DATA = [{
  name: "Row Single Line Headline Caption",
  tag: "RowHeadlineImageCaption",
  description: "A row with left aligned headline text and right aligned caption text",
  category: _types.COMPONENT_TYPES.row,
  preview_image_url: "{CLOUDINARY_URL}/Row_SingleLine_HeadlineCaption.png",
  supports_list_render: true,
  props: {
    title: {
      group: _types.GROUPS.data,
      label: "Title",
      description: "Headline text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "Beautiful West Coast Villa",
      editable: true,
      required: false
    },
    caption: {
      group: _types.GROUPS.data,
      label: "Caption",
      description: "Caption text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "$100",
      editable: true,
      required: false
    }
  },
  layout: {}
}, {
  name: "Row Single Line Headline Image Caption",
  tag: "RowHeadlineImageCaption",
  description: "A row with left aligned image and headline text and right aligned caption text",
  category: _types.COMPONENT_TYPES.row,
  preview_image_url: "{CLOUDINARY_URL}/Row_SingleLine_HeadlineImageCaption.png",
  supports_list_render: true,
  props: {
    title: {
      group: _types.GROUPS.data,
      label: "Title",
      description: "Headline text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "Beautiful West Coast Villa",
      editable: true,
      required: false
    },
    caption: {
      group: _types.GROUPS.data,
      label: "Caption",
      description: "Caption text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "$100",
      editable: true,
      required: false
    },
    image: {
      group: _types.GROUPS.data,
      label: "Image",
      description: "Image to display",
      formType: _types.FORM_TYPES.image,
      propType: _types.PROP_TYPES.ASSET,
      defaultValue: null,
      editable: true,
      required: false
    }
  },
  layout: {}
}, {
  name: "Row Double Line Headline Caption",
  tag: "RowHeadlineImageCaption",
  description: "A row with left aligned headline text and subtitle text and right aligned caption text",
  category: _types.COMPONENT_TYPES.row,
  preview_image_url: "{CLOUDINARY_URL}/Row_DoubleLine_HeadlineCaption.png",
  supports_list_render: true,
  props: {
    title: {
      group: _types.GROUPS.data,
      label: "Title",
      description: "Headline text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "Beautiful West Coast Villa",
      editable: true,
      required: false
    },
    subtitle: {
      group: _types.GROUPS.data,
      label: "Subtitle",
      description: "Subtitle text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "San Diego",
      editable: true,
      required: false
    },
    caption: {
      group: _types.GROUPS.data,
      label: "Caption",
      description: "Caption text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "$100",
      editable: true,
      required: false
    }
  },
  layout: {}
}, {
  name: "Row Double Line Headline Image Caption",
  tag: "RowHeadlineImageCaption",
  description: "A row with left aligned image, headline text, and subtitle text, and right aligned caption text",
  category: _types.COMPONENT_TYPES.row,
  preview_image_url: "{CLOUDINARY_URL}/Row_DoubleLine_HeadlineCaption.png",
  supports_list_render: true,
  props: {
    title: {
      group: _types.GROUPS.data,
      label: "Title",
      description: "Headline text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "Beautiful West Coast Villa",
      editable: true,
      required: false
    },
    subtitle: {
      group: _types.GROUPS.data,
      label: "Subtitle",
      description: "Subtitle text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "San Diego",
      editable: true,
      required: false
    },
    caption: {
      group: _types.GROUPS.data,
      label: "Caption",
      description: "Caption text to display",
      formType: _types.FORM_TYPES.string,
      propType: _types.PROP_TYPES.STRING,
      defaultValue: "$100",
      editable: true,
      required: false
    },
    image: {
      group: _types.GROUPS.data,
      label: "Image",
      description: "Image to display",
      formType: _types.FORM_TYPES.image,
      propType: _types.PROP_TYPES.ASSET,
      defaultValue: null,
      editable: true,
      required: false
    }
  },
  layout: {}
}];
exports.SEED_DATA = SEED_DATA;
