// @draftbit/core/lib/commonjs/components/Touchable.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Touchable;
exports.SEED_DATA = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _types = require("@draftbit/types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function Touchable({
  children,
  disabled,
  onPress,
  style,
  ...props
}) {
  return /*#__PURE__*/_react.default.createElement(_reactNative.Pressable, _extends({
    onPress: onPress,
    disabled: disabled,
    hitSlop: 8,
    style: ({
      pressed
    }) => {
      return [{
        opacity: pressed || disabled ? 0.75 : 1
      }, style];
    }
  }, props), children);
}

const SEED_DATA = {
  name: "Touchable",
  tag: "Touchable",
  description: "Simple button with no styles",
  category: _types.COMPONENT_TYPES.button,
  layout: {},
  props: {
    onPress: (0, _types.createActionProp)()
  }
};
exports.SEED_DATA = SEED_DATA;
